packages/dev-server-rollup/src/rollupAdapter.ts

```diff
@@ -192,6 +192,12 @@ export function rollupAdapter(
           const filename = path.basename(
             resolvedImportPath.replace(/\0*/g, '').split('?')[0].split('#')[0],
           );
+          // if the resolve import path is outside our normal root, fully resolve the file path for rollup
+          const matches = resolvedImportPath.match(/\/__wds-outside-root__\/([0-9]+)\/(.*)/);
```

Comment:
Can we make the regexp a constant? that avoids recreation the regexp each time, and this is quite a hot path

Reply:
Good catch, done in f02a638